Source/WebKit/UIProcess/API/glib/WebKitInitialize.cpp

```diff
@@ -75,8 +75,15 @@ static void initializeRemoteInspectorServer()
 
     auto inspectorHTTPAddress = parseAddress(httpAddress);
     GRefPtr<GSocketAddress> inspectorAddress;
-    if (inspectorHTTPAddress)
-        inspectorAddress = adoptGRef(G_SOCKET_ADDRESS(g_inet_socket_address_new(g_inet_socket_address_get_address(G_INET_SOCKET_ADDRESS(inspectorHTTPAddress.get())), 0)));
+    if (inspectorHTTPAddress) {
+        const char* inspectorPortStr = g_getenv("WEBKIT_INSPECTOR_PORT");
+        auto port = inspectorPortStr ? g_ascii_strtoull(inspectorPortStr, nullptr, 10) : 0;
+        if (port > std::numeric_limits<uint16_t>::max()) {
+            g_warning("WEBKIT_INSPECTOR_PORT is out of range: %s. Falling back to dynamic port", inspectorPortStr);
+            port = 0;
+        }
```

Comment:
This should be replaced with:

String envVar = String::fromLatin1(getenv("WEBKIT_INSPECTOR_PORT"));
auto port = parseInteger<uint16_t>(envVar).value_or(0);

You can check whether port is 0 after this if you want to log the warning that we're falling back to a dynamic port.

You may need to include <wtf/text/StringToIntegerConversion.h> for it to work.

Reply:
Thanks, done.
port will also be 0 if env does not exist or if it's set to "0". Lets skip the log, it's unlikely anyway